packages/iov-ethereum/src/ethereumconnection.spec.ts

```diff
@@ -1017,19 +1113,244 @@ describe("EthereumConnection", () => {
+          kind: "bcp/send",
+          creator: sender,
+          recipient: recipientAddress,
+          fee: {
+            gasPrice: testConfig.gasPrice,
+            gasLimit: testConfig.gasLimit,
+          },
+          ...testConfig.erc20TransferTests[0],
+        };
+
+        const sendB: SendTransaction = {
+          kind: "bcp/send",
+          creator: sender,
+          recipient: recipientAddress,
+          fee: {
+            gasPrice: testConfig.gasPrice,
+            gasLimit: testConfig.gasLimit,
+          },
+          ...testConfig.erc20TransferTests[0],
+        };
+
+        const sendC: SendTransaction = {
+          kind: "bcp/send",
+          creator: sender,
+          recipient: recipientAddress,
+          fee: {
+            gasPrice: testConfig.gasPrice,
+            gasLimit: testConfig.gasLimit,
+          },
+          ...testConfig.erc20TransferTests[0],
+        };
+
+        const [nonceA, nonceB, nonceC] = await connection.getNonces({ pubkey: sender.pubkey }, 3);
+
+        const signedA = await profile.signTransaction(sendA, codec, nonceA);
+        const signedB = await profile.signTransaction(sendB, codec, nonceB);
+        const signedC = await profile.signTransaction(sendC, codec, nonceC);
+        const bytesToPostA = codec.bytesToPost(signedA);
+        const bytesToPostB = codec.bytesToPost(signedB);
+        const bytesToPostC = codec.bytesToPost(signedC);
+
+        const transactionIds = new Set<TransactionId>();
+
+        // Post A and B, wait for a block
+        const postResultA = await connection.postTx(bytesToPostA);
+        const postResultB = await connection.postTx(bytesToPostB);
+        transactionIds.add(postResultA.transactionId);
+        transactionIds.add(postResultB.transactionId);
+        await postResultA.blockInfo.waitFor(info => !isBlockInfoPending(info));
+        await postResultB.blockInfo.waitFor(info => !isBlockInfoPending(info));
+
+        // setup listener after A and B are in block
+        const events = new Array<ConfirmedTransaction>();
+        const subscription = connection.liveTx({ sentFromOrTo: recipientAddress }).subscribe({
+          next: event => {
+            if (!isConfirmedTransaction(event)) {
+              throw new Error("Confirmed transaction expected");
+            }
+
+            events.push(event);
+
+            if (!isSendTransaction(event.transaction)) {
+              throw new Error("Unexpected transaction type");
+            }
+            expect(event.transaction.recipient).toEqual(recipientAddress);
+            expect(event.transaction.creator).toEqual(sender);
+            expect(event.primarySignature.pubkey).toEqual(sender.pubkey);
+
+            if (events.length === 3) {
+              const receivedIds = new Set(events.map(e => e.transactionId));
+              expect(receivedIds).toEqual(transactionIds);
+
+              subscription.unsubscribe();
+              connection.disconnect();
+              done();
```

Comment:
Did you consider doing this Promise-style?

Reply:
Wrapping those event-based tests into promises hides a lot of the underlying detail and removes control, which is important for debugging. Thus all the liveTx/listenTx tests are written this way.